### PR TITLE
[testcase] add groupby & limit test cases

### PR DESCRIPTION
**Pull request by jude-zhu**

### CODE REVIEW

**Comment by jude-zhu** [comment]
not ready for testing 
a related bug #527

**Review by nevermore3 — DISMISSED** [review]

File: tests/tck/features/go/GroupbyLimit.IntVid.feature [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      GO FROM hash("Marco Belinelli") OVER serve YIELD $$.team.name AS name, COUNT(serve._dst) AS id
+      """
+    Then a SemanticError should be raised at runtime:
+
+  Scenario: Limit test
+    When executing query:
+      """
+      GO FROM hash("Marco Belinelli") OVER serve YIELD $$.team.name AS name | ORDER BY $-.name | LIMIT 5
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.IntVid.feature:81]: in order

File: tests/tck/features/go/GroupbyLimit.IntVid.feature [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      | "76ers"   |
+      | "Bulls"   |
+      | "Hawks"   |
+      | "Hornets" |
+      | "Hornets" |
+    When executing query:
+      """
+      GO FROM hash("Marco Belinelli") OVER serve YIELD $$.team.name AS name | ORDER BY $-.name | LIMIT 2,2
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.IntVid.feature:92]: in order

File: tests/tck/features/go/GroupbyLimit.IntVid.feature [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      | name      |
+      | "Hawks"   |
+      | "Hornets" |
+    When executing query:
+      """
+      GO FROM hash("Marco Belinelli") OVER serve YIELD $$.team.name AS name | ORDER BY $-.name | LIMIT 2 OFFSET 2
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.IntVid.feature:100]: ditto

File: tests/tck/features/go/GroupbyLimit.IntVid.feature [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+      | name      |
+      | "Hawks"   |
+      | "Hornets" |
+    When executing query:
+      """
+      GO FROM hash("Marco Belinelli") OVER like YIELD $$.team.name AS name, like._dst AS id | ORDER BY $-.name | LIMIT 1
+      | GO FROM $-.id OVER like YIELD $$.player.name AS name | ORDER BY $-.name | LIMIT 2
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.IntVid.feature:109]: ditto

File: tests/tck/features/go/GroupbyLimit.IntVid.feature [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+      | "Chris Paul"      | 66          | 33.0        | 33          | 33          | 1       | 2      | 0       | 2                  | 1                           |
+      | "LeBron James"    | 68          | 34.0        | 34          | 34          | 1       | 2      | 0       | 2                  | 1                           |
+
+  Scenario: Groupby works with orderby or limit test
+    When executing query:
+      """
+      GO FROM hash("Carmelo Anthony"),hash("Dwyane Wade") OVER like YIELD $$.player.name AS name | GROUP BY $-.name, abs(5)
+      YIELD $-.name AS name, SUM(1.5) AS sum, COUNT(*) AS count | ORDER BY $-.sum, $-.name
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.IntVid.feature:209]: ditto

File: tests/tck/features/go/GroupbyLimit.feature [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+      | "Chris Paul"      | 66          | 33.0        | 33          | 33          | 1       | 2      | 0       | 2                  | 1                           |
+      | "LeBron James"    | 68          | 34.0        | 34          | 34          | 1       | 2      | 0       | 2                  | 1                           |
+
+  Scenario: Groupby works with orderby or limit test
+    When executing query:
+      """
+      GO FROM "Carmelo Anthony","Dwyane Wade" OVER like YIELD $$.player.name AS name | GROUP BY $-.name, abs(5)
+      YIELD $-.name AS name, SUM(1.5) AS sum, COUNT(*) AS count | ORDER BY $-.sum, $-.name
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.feature:216]: in order

File: tests/tck/features/go/GroupbyLimit.feature [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+      | "Carmelo Anthony" | 1.5 | 1     |
+      | "Dwyane Wade"     | 1.5 | 1     |
+      | "Chris Paul"      | 3.0 | 2     |
+      | "LeBron James"    | 3.0 | 2     |
+    When executing query:
+      """
+      GO FROM "Carmelo Anthony","Dwyane Wade" OVER like YIELD $$.player.name AS name | GROUP BY $-.name, abs(5)
+      YIELD $-.name AS name, SUM(1.5) AS sum, COUNT(*) AS count | ORDER BY $-.sum, $-.name DESC | LIMIT 2
+      """
+    Then the result should be, in any order, with relax comparison:
```
Comment (yixinglu) [inline tests/tck/features/go/GroupbyLimit.feature:227]: ditto

**Review by yixinglu — APPROVED** [review]

**Comment by czpmango** [comment]
LGTM.

**Review by czpmango — APPROVED** [review]

**Pull request merged**